src/EditorFeatures/CSharp/CodeCleanup/CSharpCodeCleanupService.cs

```diff
@@ -0,0 +1,181 @@
+                                          IDEDiagnosticIds.UseExpressionBodyForPropertiesDiagnosticId,
+                                          IDEDiagnosticIds.UseExpressionBodyForIndexersDiagnosticId,
+                                          IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId)
+                },
+                {
+                    FeatureOnOffOptions.FixInlineVariableDeclarations,
+                    ImmutableArray.Create(IDEDiagnosticIds.InlineDeclarationDiagnosticId)
+                },
+                {
+                    FeatureOnOffOptions.RemoveUnusedVariables,
+                    ImmutableArray.Create(IDEDiagnosticIds.DeclaredVariableNeverUsedDiagnosticId,
+                                          IDEDiagnosticIds.AssignedVariableNeverUsedDiagnosticId)
+                },
+                {
+                    FeatureOnOffOptions.FixObjectCollectionInitialization,
+                    ImmutableArray.Create(IDEDiagnosticIds.UseObjectInitializerDiagnosticId,
+                                          IDEDiagnosticIds.UseCollectionInitializerDiagnosticId)
+                }
+            };
+
+            return dictionary.ToImmutableDictionary();
+        }
+
+        public async Task<Document> CleanupDocument(Document document, CancellationToken cancellationToken)
+        {
+            var docOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+
+            document = await ApplyCodeFixes(document, docOptions, cancellationToken).ConfigureAwait(false);
+            // do the remove usings after code fix, as code fix might remove some code which can results in unused usings.
+            document = await RemoveSortUsings(document, docOptions, cancellationToken).ConfigureAwait(false);
+
+            return document;
+        }
+
+        private async Task<Document> RemoveSortUsings(Document document, DocumentOptionSet docOptions, CancellationToken cancellationToken)
+        {
+            // remove usings
+            if (docOptions.GetOption(FeatureOnOffOptions.RemoveUnusedUsings))
+            {
+                var removeUsingsService = document.GetLanguageService<IRemoveUnnecessaryImportsService>();
+                if (removeUsingsService != null)
+                {
+                    document = await removeUsingsService.RemoveUnnecessaryImportsAsync(document, cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            // sort usings
+            if (docOptions.GetOption(FeatureOnOffOptions.SortUsings))
+            {
+                document = await OrganizeImportsService.OrganizeImportsAsync(document, cancellationToken).ConfigureAwait(false);
+            }
+
+            return document;
+        }
+
+        private async Task<Document> ApplyCodeFixes(Document document, DocumentOptionSet docOptions, CancellationToken cancellationToken)
```

Comment:
Task returning methods should end in "Async".

Reply:
fixed